sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1926,6 +1926,23 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val MAX_COMMON_EXPRS_IN_COLLAPSE_PROJECT =
+    buildConf("spark.sql.optimizer.maxCommonExprsInCollapseProject")
+      .doc("An integer number indicates the maximum allowed number of common input expression " +
+        "from lower Project when being collapsed into upper Project by optimizer rule " +
+        "`CollapseProject`. Normally `CollapseProject` will collapse adjacent Project " +
```

Comment:
(Just a comment) If we set `CollapseProject` in `spark.sql.optimizer.excludedRules`, it seems like Spark still respects this value in `ScanOperation`? That behaviour might be okay, but it looks a bit weird to me.

Reply:
Yeah, but currently if we exclude `CollapseProject`, `ScanOperation` will work and collapse projections. Maybe update this doc?

Reply:
hm I see. Yea, updating the doc sounds nice to me.